### PR TITLE
Make Search feature available via MinIO operator

### PR DESCRIPTION
**Pull request by krisis**
It comprises of k8s secret to hold postgres creds (internal), a statefulset for
running postgres server and headless service to make it accessible to the
tenant's MinIO.

Requires #354

### CODE REVIEW

**Comment by krisis** [comment]
log_feature_arch_1
Above image captures the different components that go into making the log-search feature.

**Review by donatello — DISMISSED** [review]

File: examples/tenant-log-search.yaml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    organizationName: []
+    dnsNames: []
+
+  ## PodManagement policy for MinIO Tenant Pods. Can be "OrderedReady" or "Parallel"
+  ## Refer https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/#pod-management-policy
+  ## for details.
+  podManagementPolicy: Parallel
+
+  log:
+    image: minio/logsearchapi:v3.0.28-14-gf0a7b7a
```
Comment (krisis) [inline examples/tenant-log-search.yaml:101]: The container image version will be determined based on the operator (released) version after https://github.com/minio/operator/pull/354 is merged. Until then, this is a placeholder.

File: pkg/controller/cluster/log-search.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	conn, err := pgx.Connect(ctx, ld.connStr)
+	if err != nil {
+		return err
+	}
+	conn.Close(ctx) //nolint:errcheck
+	return nil
+}
+
+// CreateAuditDB creates the audit log database in tenant's DB.
+func (ld *logDB) CreateAuditDB(ctx context.Context) error {
```
Comment (dvaldivia) [inline pkg/controller/cluster/log-search.go:57]: I think this is the only code that needs the `pgx` dependency, if we are using the official postgres image we could set `POSTGRES_DB` and that would create a database with the desired name by default. This way we don't need to add the pgx and libpq dependencies to operator

**Review by dvaldivia — CHANGES_REQUESTED** [review]

**Review by Praveenrajmani — COMMENTED** [review]
LGTM overall. Just a minor comment

File: pkg/controller/cluster/log-search.go (hunk 2 above)
Comment (krisis) [reply pkg/controller/cluster/log-search.go:57]: This has been addressed in the recent update to this PR.

File: pkg/controller/cluster/main-controller.go [hunk 3]
```diff
@@ -1494,3 +1534,98 @@ func MinIOControllerRateLimiter() queue.RateLimiter {
...
+	klog.V(2).Infof("Creating a new Log secret for %s", tenant.Name)
+	secret, err = c.kubeClientSet.CoreV1().Secrets(tenant.Namespace).Create(ctx, secrets.LogSecret(tenant), metav1.CreateOptions{})
+	return secret, err
+}
+
+func (c *Controller) checkAndConfigureLogSearchAPI(ctx context.Context, tenant *miniov1.Tenant, secret *corev1.Secret, adminClnt *madmin.AdminClient) error {
+	// Check if audit webhook is configured for tenant's MinIO
+	auditCfg := newAuditWebhookConfig(tenant, secret)
+	_, err := adminClnt.GetConfigKV(ctx, auditCfg.target)
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (krisis) [reply pkg/controller/cluster/main-controller.go:1608]: @Praveenrajmani pkg/madmin package doesn't export such error values. In this case, the operator uses the absence of audit webhook configuration to indicate that tenant MinIO's audit webhook hasn't been configured yet.

File: examples/tenant-log-search.yaml (hunk 1 above)
Comment (krisis) [reply examples/tenant-log-search.yaml]: This is fixed now. `minio/logsearchapi` is now available in docker hub.

**Review by Praveenrajmani — DISMISSED** [review]

**Comment by krisis** [comment]
Rebased and tested on top of latest master. PTAL @donatello @Praveenrajmani

**Review by Praveenrajmani — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**